File: numberPattern2.java
/**
 * The NumberPattern2 class prints a pattern of numbers in descending order for a given number of rows
 * and columns.
 */
class NumberPattern2 {
    public static void main(String[] args) {
        int raw = 4;
        int column =4;
        String s ="";
        for (int i = 1; i <= raw; i++) {
            for (int j =column; j >= 1; j--){
                s +=j;
            }
            s+="\n";
        }
        System.out.println(s);
   }
}

File: numberPattern2.ts
/* This code is creating a string of numbers in a grid format with 4 rows and 4 columns. It first
initializes the number of rows and columns using `const raw = 4;` and `const column = 4;`. It then
uses a nested loop to iterate through each row and column and add the column number to a string
variable `string`. After each row is complete, it adds a newline character `\n` to the string to
move to the next row. Finally, it logs the resulting string to the console using
`console.log(string);`. */
const raw = 4;
const column =4;
let string = ''
for(let i=1;i<=raw;i++){
   for(let j=1;j<=column;j++){
    string +=j
}
string += "\n"
}
console.log(string);